packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -0,0 +1,251 @@
+  private _dirtyFlag: number;
+
+  /**
+   * The reference to the used texture.
+   */
+  get texture(): Texture2D {
+    return this._texture;
+  }
+
+  set texture(value: Texture2D) {
+    if (this._texture !== value) {
+      this._texture = value;
+    }
+  }
+
+  /**
+   * The rectangle this sprite uses on its texture.
+   */
+  get textureRect(): Rect {
+    return this._textureRect;
+  }
+
+  set textureRect(value: Rect) {
+    if (this._textureRect !== value) {
+      this._textureRect.x = value.x;
+      this._textureRect.y = value.y;
+      this._textureRect.width = value.width;
+      this._textureRect.height = value.width;
+    }
+  }
+
+  /**
+   * TODO
+   * The border sizes of the sprite.
+   * @remarks
+   * x=left, y=right, z=top, w=bottom.
+   */
+  get border(): Vector4 {
+    return this._border;
+  }
+
+  set border(border: Vector4) {
+    this._border = border;
+  }
+
+  /**
+   * Location of the sprite's center point in the rect on the original texture, specified in pixels.
+   */
+  get pivot(): Vector2 {
+    return this._pivot;
+  }
+
+  set pivot(value: Vector2) {
+    if (this._pivot !== value) {
+      value.cloneTo(this._pivot);
+      this._setDirtyFlagTrue(Sprite._VERTICES_FLAG);
+    }
+  }
+
+  /**
+   * Location of the sprite on the original texture, specified in pixels.
+   */
+  get rect(): Rect {
+    return this._rect;
+  }
+
+  set rect(value: Rect) {
+    if (this._rect !== value) {
+      this._rect.x = value.x;
+      this._rect.y = value.y;
+      this._rect.width = value.width;
+      this._rect.height = value.width;
+      this._setDirtyFlagTrue(Sprite._VERTICES_FLAG);
+    }
+  }
+
+  /**
+   * The number of pixels in the sprite that correspond to one unit in world space.
+   */
+  get pixelsPerUnit(): number {
+    return this._pixelsPerUnit;
+  }
+
+  set pixelsPerUnit(value: number) {
+    if (this._pixelsPerUnit !== value) {
+      this._pixelsPerUnit = value;
+      this._setDirtyFlagTrue(Sprite._VERTICES_FLAG);
+    }
+  }
+
+  /**
+   * Constructor a sprite.
+   * @param engine - Engine to which the sprite belongs
+   * @param texture - Texture from which to obtain the sprite
+   * @param rect - Rectangular section of the texture to use for the sprite
+   * @param pivot - Sprite's pivot point relative to its graphic rectangle
+   * @param pixelsPerUnit - The number of pixels in the sprite that correspond to one unit in world space
+   */
+  constructor(
+    engine: Engine,
+    texture: Texture2D,
+    rect: Rect = null,
+    pivot: Vector2 = null,
+    pixelsPerUnit: number = 100
+  ) {
+    super(engine);
+
+    if (!rect) {
+      rect = new Rect(0, 0, texture.width, texture.height);
+    }
+    if (!pivot) {
+      pivot = new Vector2(0.5, 0.5);
+    }
+
+    if (rect.x + rect.width > texture.width || rect.y + rect.height > texture.height) {
+      throw new Error("rect out of range");
+    }
+
+    // Init.
+    this._dirtyFlag = Sprite._VER_UV_TRI_FLAG;
```

Comment:
const value init in construtor is not good

Reply:
> const value init in construtor is not good

done